sdk/test-utils/recorder/src/baseRecorder.ts

```diff
@@ -153,54 +153,61 @@ export class NockRecorder extends BaseRecorder {
-    }
+    if (isRecordMode()) {
+      // Importing "nock" library in the recording and appending the testInfo part in the recording
+      const importNockStatement =
+        "let nock = require('nock');\n" +
+        "\n" +
+        "module.exports.testInfo = " +
+        JSON.stringify(this.uniqueTestInfo) +
+        "\n";
+
+      const fixtures = nock.recorder.play();
+
+      // Create the directories recursively incase they don't exist
+      try {
+        // Stripping away the filename from the filepath and retaining the directory structure
+        fs.ensureDirSync(
+          "./recordings/" +
+            this.relativeTestRecordingFilePath.substring(
+              0,
+              this.relativeTestRecordingFilePath.lastIndexOf("/") + 1
+            )
+        );
+      } catch (err) {
+        if (err.code !== "EEXIST") throw err;
+      }
 
-    const file = fs.createWriteStream("./recordings/" + this.relativeTestRecordingFilePath, {
-      flags: "w"
-    });
+      const file = fs.createWriteStream("./recordings/" + this.relativeTestRecordingFilePath, {
+        flags: "w"
+      });
 
-    // Some tests expect errors to happen and, if a writing error is thrown in one of these tests, it may be captured in a catch block by accident,
-    // resulting in unexpected behavior. For this reason we're printing it to the console as well
-    file.on("error", (err: any) => {
-      console.log(err);
-      throw err;
-    });
+      // Some tests expect errors to happen and, if a writing error is thrown in one of these tests, it may be captured in a catch block by accident,
+      // resulting in unexpected behavior. For this reason we're printing it to the console as well
+      file.on("error", (err: any) => {
+        console.log(err);
+        throw err;
+      });
 
-    file.write(importNockStatement);
+      file.write(importNockStatement);
 
-    // Saving the recording to the file
-    for (const fixture of fixtures) {
-      // We're not matching query string parameters because they may contain sensitive information, and Nock does not allow us to customize it easily
-      const updatedFixture = fixture.toString().replace(/\.query\(.*\)/, ".query(true)");
-      file.write(this.filterSecrets(updatedFixture) + "\n");
-    }
+      // Saving the recording to the file
+      for (const fixture of fixtures) {
+        // We're not matching query string parameters because they may contain sensitive information, and Nock does not allow us to customize it easily
+        const updatedFixture = fixture.toString().replace(/\.query\(.*\)/, ".query(true)");
+        file.write(this.filterSecrets(updatedFixture) + "\n");
+      }
 
-    file.end();
+      file.end();
 
-    nock.recorder.clear();
-    nock.restore();
+      nock.recorder.clear();
+      nock.restore();
+    } else if (isPlaybackMode()) {
+      nock.restore();
+      nock.cleanAll();
+      nock.enableNetConnect();
+      nock.activate();
```

Comment:
can we move activate to the moment when nock is first loaded? Like this: https://github.com/Azure/azure-sdk-for-js/pull/7254/files#diff-0dbc79673f7fccf741ddfba1117eea3aR45

A comment might also help understand why this is needed.

Reply:
Updated.